src/basic/REDUCE_STRUCT.hpp

```diff
@@ -0,0 +1,134 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2017-21, Lawrence Livermore National Security, LLC
+// and RAJA Performance Suite project contributors.
+// See the RAJAPerf/LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+///
+/// REDUCE_STRUCT kernel reference implementation:
+///
+/// Real_type xsum = 0.0;
+/// Real_type xmin = 0.0;
+/// Real_type xmax = 0.0;
```

Comment:
Missing y var declarations.
If the min is initialized to 0, it can never find a min greater than 0.